packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,283 @@
+
+// Styles
+import viewSettingsDialogCSS from "./generated/themes/ViewSettingsDialog.css.js";
+
+const metadata = {
+	tag: "ui5-view-settings-dialog",
+	managedSlots: true,
+	slots: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the <code>Sort By</code> list.
+		 * @type {sap.ui.webcomponents.fiori.List}
+		 * @slot sortBy
+		 * @public
+		 */
+		 "sortBy": {
+			propertyName: "sortBy",
+			type: HTMLElement,
+		},
+	},
+	properties: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the initial sort order.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 sortDescending: {
+			type: Boolean,
+		},
+	},
+	events: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+
+		/**
+		 * Fired when OK button is activated.
+		 *
+		 * @event sap.ui.webcomponents.fiori.ViewSettingsDialog#ok-click
+		 * @param {String} sortOrder The current sort order selected.
+		 * @param {String} sortBy The current sort by selected.
+		 * @public
+		 */
+		"ok-click": {
+			detail: {
+				sortOrder: { type: String },
+				sortBy: { type: String },
+			},
+		},
+		/**
+		 * Fired when Cancel button is activated.
+		 *
+		 * @event sap.ui.webcomponents.fiori.ViewSettingsDialog#cancel-click
+		 * @param {Object} settings The current settings.
+		 * @public
+		 */
+		"cancel-click": {},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-view-settings-dialog</code> component consisting of several lists.
+ * One of them (<code>Sort order</code>) is built-in, and another (<code>Sort By</code>) must be provided by the developer.
+ * The selected options can be used to create sorters for the table.
+ *
+ * The <code>ui5-view-settings-dialog</code> interrupts the current app processing as it is the only focused UI element and
+ * the main screen is dimmed/blocked.
+ * The <code>ui5-view-settings-dialog</code> is modal, which means that user action is required before returning to the parent window is possible.
+ *
+ * <h3>Structure</h3>
+ * A <code>ui5-view-settings-dialog</code> consists of a header, content, and a footer for action buttons.
+ * The <code>ui5-view-settings-dialog</code> is usually displayed at the center of the screen.
+ *
+ * <h3>Responsive Behavior</h3>
+ * <code>ui5-view-settings-dialog</code> stretches on full screen on phones.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/ViewSettingsDialog";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.ViewSettigsDialog
+ * @extends UI5Element
+ * @tagname ui5-view-settings-dialog
+ * @public
+ */
+class ViewSettingsDialog extends UI5Element {
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
+		this._initialSettings = {}; // settings when the control is opened for the first time
```

Comment:
IMO the variable names are self explanatory and we don't need the comments

Reply:
done